ext/mbstring/mbstring.c

```diff
@@ -4135,20 +4108,16 @@ static inline zend_string *php_mb_chr(zend_long cp, zend_string *enc_name, uint3
 	buf[2] = (cp >>  8) & 0xff;
 	buf[3] = cp & 0xff;
 
-	size_t ret_len;
 	long orig_illegalchars = MBSTRG(illegalchars);
 	MBSTRG(illegalchars) = 0;
-	char *ret_str = php_mb_convert_encoding_ex(buf, 4, enc, &mbfl_encoding_ucs4be, &ret_len);
+	ret = php_mb_convert_encoding_ex(buf, 4, enc, &mbfl_encoding_ucs4be);
+
 	if (MBSTRG(illegalchars) != 0) {
-		efree(ret_str);
-		MBSTRG(illegalchars) = orig_illegalchars;
-		return NULL;
+		efree(ret);
```

Comment:
This should use zend_string_release.

Reply:
Thanks for the catch. I didn't know freeing a `zend_string` was different from any other malloc'd object. Looking at the source, it makes sense.